src/server/typingsInstaller/typingsInstaller.ts

```diff
@@ -332,6 +338,9 @@ namespace ts.server.typingsInstaller {
                     if (this.log.isEnabled()) {
                         this.log.writeLine(`Installed typings ${JSON.stringify(scopedTypings)}`);
                     }
+
+                    this.writePackageVersion(cachePath);

```

Comment:
- We only want to write out the version after we've actually installed types. It's too early to do it after ```processCacheLocation```, process may terminate before actually installing anything

- We don't ever want to wipe out devDependencies and force a full cache refresh. I'm trying to scope the refreshes to only packages whose ```typeScriptVersions``` are truly incompatible. I expect there'll be users with, say a 2.2 project and a 2.3 project who keep switching back and forth between the two in VS, and wiping out the cache upon seeing the version change would mean that they lose the typings until they go online again. Which seems pretty bad.